subprojects/execution/src/test/groovy/org/gradle/internal/execution/history/impl/DefaultOutputFilesRepositoryTest.groovy

```diff
@@ -72,4 +74,9 @@ class DefaultOutputFilesRepositoryTest extends Specification {
         tmpDir.file(path).absoluteFile
     }
 
+    private FileSystemLocationSnapshot snapshot(File file) {
+        def builder = MerkleDirectorySnapshotBuilder.sortingRequired()
+        virtualFileSystem.read(file.getAbsolutePath(), builder)
+        return builder.getResult()
+    }
```

Comment:
Is it a problem that we now have to determine whether sorting is required at the call site instead of in the snapshotter?

Reply:
Currently, all the usages want the `FileSystemLocationSnapshot`. I wonder if we should return the snapshot from the `read` methods directly instead of passing in the visitor. WDYT? This would also allow to remove the need for `MerkleDirectorySnapshotBuilder.resultKnown()`.

Reply:
But would disallow guarding a piece of code that reads stuff. We could extract the code to a separate class instead.

But my question was a bit different. Previously caller code wasn't aware whether the snapshots would be sorted, because this was decided inside the `FileSystemSnapshotter`. Now it has to be specified at every call site. What do we think about this?

Reply:
Now the `read()` methods take a `Consumer<FileSystemLocationSnapshot`, so the given problem does not exist any more.